libcontainer/rootfs_linux.go

```diff
@@ -719,19 +730,23 @@ func checkProcMount(rootfs, dest, source string) error {
 	}
 	if path == "." {
 		// an empty source is pasted on restore
-		if source == "" {
+		if m == nil {
 			return nil
 		}
-		// only allow a mount on-top of proc if it's source is "proc"
-		isproc, err := isProc(source)
-		if err != nil {
-			return err
-		}
-		// pass if the mount is happening on top of /proc and the source of
-		// the mount is a proc filesystem
-		if isproc {
+		// only allow a mount on-top of proc if its source is a procfs mount ...
+		if m.Device == "proc" {
```

Comment:
Why change the logic here in this commit?

Reply:
1. The old check was incorrect (the source path is ignored for proper mounts of pseudo-filesystems like `tmpfs` or `sysfs`, so checking the filesystem type of the source only makes sense if it's a bind-mount).
2. We need to use `srcStatfs` because of the `/proc/thread-self` stuff.

I can make a separate commit fixing (1).